zerver/lib/streams.py

```diff
@@ -1132,3 +1132,13 @@ def get_subscribed_private_streams_for_user(user_profile: UserProfile) -> QueryS
         .filter(subscribed=True)
     )
     return subscribed_private_streams
+
+
+def get_stream_deletion_event(streams: list[Stream]) -> dict[str, Any]:
```

Comment:
I think would be good to just send the event in this function only and name this `send_stream_deletion_event` accepting the required parameters.

Reply:
ok, but I think it's a good idea to still construct the event in this function, this prevents repeating the following code : 
```python
dict(
        type="stream",
        op="delete",
        # "streams" is deprecated, kept only for compatibility.
        streams=[dict(stream_id=stream.id) for stream in streams],
        stream_ids=[stream.id for stream in streams],
    )
```

Reply:
`send_event_on_commit` already handles sending the event

Reply:
I meant just doing all this work including constructing the event and calling `send_event_on_commit` in the new function like we do in `send_stream_creation_event`.

Reply:
Ok